### PR TITLE
cleanup(rules): transition rule `BPF Program Not Profiled` to maturity incubating

### PR DESCRIPTION
**Pull request by incertum**
…y incubating

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind feature

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

> /area maturity-stable

/area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:

cleanup(rules): transition rule `BPF Program Not Profiled` to maturity incubating as it is considered a critical baseline detection.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: rules/falco-incubating_rules.yaml [hunk 1]
```diff
@@ -1284,3 +1284,24 @@
...
+
+- macro: bpf_profiled_procs
+  condition: (proc.name in (bpf_profiled_binaries))
+
+- rule: BPF Program Not Profiled
+  desc: > 
+    BPF is a kernel technology that can be misused for malicious purposes, like "Linux Kernel Module Injection". This 
+    rule should be considered an auditing rule to notify you of any unprofiled BPF tools running in your environment. 
+    However, it requires customization after profiling your environment. BPF-powered agents make bpf syscalls all the 
+    time, so this rule only sends logs for BPF_PROG_LOAD calls (bpf cmd=BPF_PROG_LOAD) in the enter event. If you also want to log
```
Comment (incertum) [inline rules/falco-incubating_rules.yaml:1299]: also fixed docs as it's now `cmd=BPF_PROG_LOAD` and not `5` anymore.

File: rules/falco-incubating_rules.yaml [hunk 2]
```diff
@@ -1283,4 +1283,25 @@
     (fd.name contains "liblzma.so.5.6.0" or fd.name contains "liblzma.so.5.6.1")
   output: SSHD loaded a backdoored version of liblzma library %fd.name with parent %proc.pname and cmdline %proc.cmdline (process=%proc.name parent=%proc.pname file=%fd.name evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid proc_exepath=%proc.exepath command=%proc.cmdline terminal=%proc.tty exe_flags=%evt.arg.flags %container.info)
   priority: WARNING
-  tags: [maturity_incubating, host, container]
+  tags: [maturity_incubating, host, container, mitre_initial_access, T1556]
```
Comment (incertum) [inline rules/falco-incubating_rules.yaml:1286]: @darryk10 @loresuso @LucaGuerra

File: rules/falco-incubating_rules.yaml (hunk 1 above)
Comment (LucaGuerra) [reply rules/falco-incubating_rules.yaml:1299]: This requires an engine version bump if I'm not mistaken, meaning also a major version bump for the incubating file

Comment (incertum) [reply rules/falco-incubating_rules.yaml:1299]: Right, forgot about it, great callout Luca!

**Review by darryk10 — APPROVED** [review]
Hi @incertum, totally agree with this change! 
LGTM

**Review by LucaGuerra — APPROVED** [review]

**Pull request merged**